lib/entry/crossDCsStreamPersistence.js

```diff
@@ -0,0 +1,209 @@
+ * Created by lilach.maliniak on 18/09/2016.
+ */
+var persistenceFormat = require('./../../common/PersistenceFormat');
+var qio = require('q-io/fs');
+//var sessionManager = require('./../SessionManager');
+var logger = require('../../common/logger').getLogger("CrossDCsStreamPersistence");
+var config = require('./../../common/Configuration');
+var ErrorUtils = require('./../utils/error-utils');
+var _ = require('underscore');
+var Q = require('q');
+var glob = require('glob');
+var path = require('path');
+var backendClient = require('../BackendClientFactory.js').getBackendClient();
+var PlaylistGenerator = require('./../playlistGenerator/PlaylistGenerator');
+const endListPendingEntriesFileName = "persistenceList.json";
+const persistenceListPath = config.get('persistenceListPath');
+const kalturaTypes = require('./../kaltura-client-lib/KalturaTypes');
+const presentationEndTimeRegex = /("presentationEndTime":\s*)(\d+)/;
+const expireTimeRegex = /("expirationTime":\s*)(\d+)/;
+
+
+function dateTimeStringFormatter(time) {
+    return new Date(time).toString().replace(/\S+\s(\S+)\s(\d+)\s(\d+)\s(\S+)\s.*/, '$2‌​-$1-$3 $4');
+}
+
+function playlistPath(entryId) {
+    return path.join(persistenceFormat.getEntryBasePath(entryId), persistenceFormat.getMasterManifestName());
+}
+
+class CrossDCsStreamPersistence {
+
+    constructor() {
+
+        this.stoppedEntriesFileName = path.join(persistenceListPath, endListPendingEntriesFileName);
+        this._persistenceCheckIntervalMillisec = config.get('persistenceCheckIntervalMillisec');
+        this._entriesToProcess = {};
+        this._lastProcessedTimestampe = Date.now();
+    }
+
+    streamPersistenceCheck(entryId, newEntry = false) {
+        logger.debug(`[${entryId}] starting stream persistence check`);
+
+        // todo: consult with Guy whether to filter received nodes objects according to status...
+        //entryServerNodeFilter.statusEqual = KalturaLiveStatus.PLAYABLE or KalturaLiveStatus.BROADCASTING;
+        return backendClient.getLiveEntryServerNodes(entryId)
+            .then((serverObjs) => {
+                return _.reduce(serverObjs, (memo, dc) => {
+                    return (dc.status === kalturaTypes.KalturaEntryServerNodeStatus.PLAYABLE) || memo;
+                }, false);
+            })
+            .then((isLive) => {
+                return this.playlistUpdate(entryId, isLive, newEntry);
+            })
+            .catch((err) => {
+                logger.error(`[${entryId}] persistence check failed. Error: [${ErrorUtils.error2string(err)}]`);
+                return Q.reject(err);
+            });
+
+    }
+
+    playlistUpdate(entryId, isLive, newEntry) {
+
+        let now = Date.now();
+        let presentationEndTime = now + 7200000;
+        let path = playlistPath.call(this, entryId);
+        let ended = !isLive;
+
+        return qio.read(path)
+            .then((content) => {
+                let playlist = content;
+                if (newEntry) {
```

Comment:
we don't really need newEntry param, actually you can always change the expire time  to the past?